presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java

```diff
@@ -25,31 +25,70 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 
+import javax.annotation.concurrent.Immutable;
+
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class StatisticAggregations
 {
+    // Added to preserve the order for serializing and deserializing.
+    // map in c++ worker doesn't represent insertion order.
+    private final List<VariableReferenceExpression> variables;
+    private final List<Aggregation> variableAggregations;
+    // Only used internally for quick lookup.
```

Comment:
This comment suggests that this map is used internally in this class and is not exposed externally. However, I see getAggregations() method returning this value directly. Hence, this comment is incorrect. Looking at the usages of getAggregations(), I see that these are used in aggregations and I also see that AggregationNode uses a map to specify the order of aggregations in the output. This means we have the same problem with regular aggregations as with aggregations used to collect stats. 

The issue with regular aggregations gets masked by Velox using column names instead of channels (indices) to lookup input columns in the operators. 


```
    @JsonCreator
    public AggregationNode(
            Optional<SourceLocation> sourceLocation,
            @JsonProperty("id") PlanNodeId id,
            @JsonProperty("source") PlanNode source,
            @JsonProperty("aggregations") Map<VariableReferenceExpression, Aggregation> aggregations,
            @JsonProperty("groupingSets") GroupingSetDescriptor groupingSets,
            @JsonProperty("preGroupedVariables") List<VariableReferenceExpression> preGroupedVariables,
            @JsonProperty("step") Step step,
            @JsonProperty("hashVariable") Optional<VariableReferenceExpression> hashVariable,
            @JsonProperty("groupIdVariable") Optional<VariableReferenceExpression> groupIdVariable)
```